go_modules/lib/dependabot/go_modules/update_checker.rb

```diff
@@ -64,6 +64,18 @@ def find_latest_resolvable_version
             # private git dependencies
             env = { "GOPRIVATE" => "*" }
 
+            # Note: rather than leveraging our custom getUpdatedVersion
+            # and then passing that into `go get` within go_mod_updater
+            # we could invert this and run `go get dep@latest` and then
+            # inspect the go.mod/go.sum file to see what it was bumped
+            # to, and then flow from there. It would remove the need for
+            # https://github.com/dependabot/gomodules-extracted.
+            # But currently `go get` is a bit slower than our fast
+            # version checker. That may change in go 1.17 with
+            # lazy-loading: https://github.com/golang/go/issues/36460
+            # Also it's unclear if this would work with the pipeline
+            # approach of dependabot which naturally leads to splitting
+            # the finding of updates from actually updating to them.
```

Comment:
So this is a much, much bigger design conversation. Def out of scope for the current PR (so possibly I should drop the code comment from this PR).

But I wanted to at least point out that there's a lot of custom code whose sole goal is to duplicate what `go get -d <dep>@latest` would do... That's the _only_ reason we need https://github.com/dependabot/dependabot-core/blob/main/go_modules/helpers/updatechecker/main.go which is the _only_ reason we need an entire repo https://github.com/dependabot/gomodules-extracted/

I realize that dependabot needs to know what the updated version was, so you'd invert the flow by then checking `go.mod` to see what the new version is and then returning it out to the code that opens the PR. And that may be to much of an inversion and simply not work with the dependabot pipeline flow. Or the perf hit from running `go get` may simply be too costly compared to running this really simple updater code (since `go get -d` not only checks for latest but also downloads the module code). But the code maintainability savings/lowered chance of bugs (at least on the `go get` logic) are quite large too. So it's a big tradeoff either way.

Reply:
> Dependabot can be configured to ignore certain versions of a dependency (e.g. >= 1.0.0).

That makes sense. I had a feeling there were going to be some use cases that make this non-tenable.

> I think something like `go list -m -mod=mod -versions -json <dep>` is a better solution than the current getUpdatedVersion helper.

Oh, I hadn't even considered that. Regardless, this is way out of scope for the current PR, so I filed https://github.com/dependabot/dependabot-core/issues/3599 and we can continue the conversation there.

I'll remove this comment from this PR.